pkg/server/container_create.go

```diff
@@ -181,8 +181,15 @@ func (c *criContainerdService) CreateContainer(ctx context.Context, r *runtime.C
 
 func (c *criContainerdService) generateContainerSpec(id string, sandboxPid uint32, config *runtime.ContainerConfig,
 	sandboxConfig *runtime.PodSandboxConfig, imageConfig *imagespec.ImageConfig, extraMounts []*runtime.Mount) (*runtimespec.Spec, error) {
+	// We use containerd spec generator for tty because we need the TERM env.
+	// TODO(random-liu): Consider whether we should completely move to containerd spec generator.
```

Comment:
I think we should pick one spec api here and make it work. I mention the runtime tools option to set the TERM env entry below. There are pros and cons to both apis. I don't have a preference for which one we go with, but yeah doing it piece meal and switching back and forth and we're probably going to miss something :-)